### ISSUE TITLE
[Prod] Agreements: Mobile users can not exit side panel after creating draft agreement without cancelling action

### ISSUE THREAD

**Issue by melyndav** [JoinColony/colonyCDapp#2933]
## Description

On mobile, when users create a draft agreement and safe it, they are forced into the modal confirming the cancellation of the agreement action to be able to exit the side panel. 

Correct behaviour:

Once the user has saved the draft action, they should be able to close the side panel without receiving the 'cancel action' confirmation modal. This should only happen on this action once the 'save draft' button has been selected. It should not impact any other action.

Desktop behaviour is working correctly. 

## Steps to reproduce (ensure you are viewing on a mobile size)

1. Log in with a MetaMask wallet
2. Open the side panel and select 'create agreement'
3. Complete the required inputs
4. Select the 'Save draft'
5. Close the side panel via the top left 'X' icon
6. The cancel action confirmation modal will show
7. Navigate the agreements page. The draft agreement should show (ensuring the save is happening). 

Step to be removed from the flow only for this action: 
<img width="466" alt="image" src="https://github.com/user-attachments/assets/081b40d7-6489-4cd6-ae03-8c942b0db72e">

## Expected behaviour

* The user should be able to save a draft agreement and exit the side panel without seeing the cancel action modal on mobile/ tablet 

## Actual behaviour

* The cancel confirmation shows after saving a draft agreement and attempting to exit the side panel which causes confusion if the draft has been saved for the user and how they can exit the side panel.

**Comment by melyndav** [comment]
@rdig can this be picked up?

**Comment by rdig** [comment]
Of course, just that realistically it won't be picked up until the end of cycle, or early next one

**Comment by melyndav** [comment]
All good, just wanted to flag it down. Thanks for adding it to cycle planning 👍 @rdig

**Comment by melyndav** [comment]
Thanks for fixing this one @rumzledz 🥳
